vpr/src/route/rr_graph_util.h

```diff
@@ -1,10 +1,31 @@
 #ifndef RR_GRAPH_UTIL_H
 #define RR_GRAPH_UTIL_H
 
+#include "vpr_types.h"
+
 int seg_index_of_cblock(t_rr_type from_rr_type, int to_node);
 
 int seg_index_of_sblock(int from_node, int to_node);
 
 void reorder_rr_graph_nodes(const t_router_opts& router_opts);
 
+// t_rr_graph_util is a utility class that collects utility
+// methods that do not strictly belong to the RR graph storage,
+// as they can use its public interface.
+class t_rr_graph_util {
```

Comment:
You don't need a class or destructor.  Just make a function that returns the vector?

Reply:
Sure, I just though that it might be possible to populate this class with additional utilities that might be necessary in the future, so to keep everything well encapsulated.

Maybe it is an overkill for now and the function is good enough.

Reply:
Done